### PR TITLE
feat: updated colors and label/button styles

### PR DESCRIPTION
**Pull request by Rukario**
- Borrow browser's color
- Pill shaped labels
- Swap peer and progress detail in torrent row (made to consistent with many other Transmission clients)
- Square priority buttons
- Even/odd rows in pure CSS

Notes: Replaced background colors with system color keywords to enable using browser's colors. CSS style adjustments esp. for label and buttons.

Left/Top: this PR, right/bottom original
Transmission light

Transmission dark

Transmission dark alt

animated diffs:

Transmission light a 2 (apng)

Transmission dark a 1 (apng)

iPhone snapshots animated

IMG_0530 (apng)

IMG_0532 (apng)

### CODE REVIEW

**Comment by GaryElshaw** [comment]
Can we please get #5762 across the line first?

**Comment by Rukario** [comment]
@GaryElshaw

I'm aware your other PR is doing color that appears to focus on color of the progress bars, I was trying to be conservative about it and do the background colors only.

**Comment by GaryElshaw** [comment]
There's some good stuff in here.

**Comment by Rukario** [comment]
@GaryElshaw

Sorry I've had to reapply my changes after a conflict, maybe my workflow isn't efficient....

**Comment by GaryElshaw** [comment]
code-style is ruthless.

**Comment by Rukario** [comment]
Is there something I can tell my setup to be strict about my coding? Is there something wrong with my code that webapp as one of the code-style checks failed at?

**Comment by GaryElshaw** [comment]
Greater minds will know better, but i think a lot of it depends on what you're using locally to write the code and the kinds of linter(s) available. Workflow wise, one file at a time, i've found, is the best path. My natural tendency is to run across multiple files simultaneously, but it just ends up making things more difficult when it comes to ci checks.

**Comment by GaryElshaw** [comment]
And always check the details of the code-style error. https://github.com/transmission/transmission/actions/runs/5861852540/job/15892673463?pr=5897

**Comment by Rukario** [comment]
I understand now, I'm sure I headed over to the details to see what exactly is failing but the plus and minus were so subtle I haven't noticed. My instinct is looking for one line it failed at. The coding rule sure is stringent but inspiring! Code looks ultra cleaning.

**Comment by tearfur** [comment]
Disclaimer: No expertise in UX whatsoever, so this is just some layman's thoughts.

> Borrow browser's color

I like the new dark mode colours more.

Just to be sure, you don't mean the colours are dependent on which browser you are using, right?

> Pill shaped labels

I like the new labels. 👍🏻 

> Even/odd rows in pure CSS

Nice!

**Comment by Rukario** [comment]
> > Borrow browser's color
> I like the new dark mode colours more.
> Just to be sure, you don't mean the colours are dependent on which browser you are using, right?

Forgive my poor word choice. The color is inherited from browser! "background: Canvas;" is clever way to do about it.

> > Pill shaped labels
> I like the new labels. 👍🏻

Looking at GitHub website layout, why not, there's some inspiration! There's some more from GitHub that can be inspired from.

**Comment by tearfur** [comment]
> Forgive my poor word choice. The color is inherited from browser!

Pardon me, for some reason I can't find the documentation for `Canvas` in this context, so I don't really know what it does.

So does that mean each browser may have different colours? If this is true, then even if it works well now on all browsers, this just seems like a feature waiting to break.

**Comment by Rukario** [comment]
Not too sure what do you mean it might break. I'm sure it's getting standardized as part of https://www.w3.org/TR/css-color-4/ but I get it there's going to be incompatibility with older browsers. I'll try implement a few more to alleviate that.

If browser doesn't understand the Canvas in the context of using for background color it won't pick up variables using them. It will be white or black depending on their appearance (light/dark).

Google can be confusing when asking about "Canvas" for HTML/CSS, it's one of system color keywords to use in CSS.

**Comment by tearfur** [comment]
> I get it there's going to be incompatibility with older browsers. I'll try implement a few more to alleviate that.

That's not what I originally meant, but yeah that is a good point too. Side note, I wonder if this can be handled by webpack by some polyfill... Similar to what Babel does to JS code.

I was worried the odd even row tint will look out of place if the browser somehow assigned a crazy colour to `Canvas`, but I now see that the way you did it is by overlaying a translucent grey layer on top, so that should be fine 👍🏻

Now I also wonder if the text should use `CanvasText`...

**Comment by ckerr** [comment]
@dareiff do you have time to review this?

**Comment by Rukario** [comment]
@tearfur

Oh, alright :) considering it's relatively new feature it's good idea to have fallback for recent browsers that don't support it, anyways. I don't know much about libraries that enable compatibility with older browsers. Do know I'm not fan of using libraries, always love vanilla approaches to anything.

> I was worried the odd even row tint will look out of place if the browser somehow assigned a crazy colour to `Canvas`, but I now see that the way you did it is by overlaying a translucent grey layer on top, so that should be fine 👍🏻

We'll have a long way until browser ever decided to use a gray background, we'll be fine!

> Now I also wonder if the text should use `CanvasText`...

That would also mean it's going to be black and white when Transmission is using non-black and non-white text that looked fine. I was just bit unsatisfied with colors Transmission is using for dark theme. Firefox has good color which I was hoping to replicate, I'm just happy there's a way to do that. I encourage using platform-provided features wherever available.

File: web/assets/css/transmission-app.scss
Comment (dareiff) [inline web/assets/css/transmission-app.scss:1]: Love these deletes; had never heard about using Canvas as you are. https://developer.mozilla.org/en-US/docs/Web/CSS/system-color

But I like it!

**Review by dareiff — APPROVED** [review]

**Comment by dareiff** [comment]
Wonderful, ship it!

**Comment by ckerr** [comment]
Thanks especially to @Rukario for the code and also everyone else in this PR for the review & feedback :green_heart:

**Comment by ckerr** [comment]
@Rukario please edit the original comment at the top of this page to add a 'Notes:' paragraph, e.,g. a paragraph that begins with `Notes:` and then has a past tense sentence describing what this PR does.

Also, a small request -- PTAL at https://www.conventionalcommits.org/en/v1.0.0/#summary and title your PRs to look like a conventional commit message, e.g. "feat: updated colors and label/button styles". I usually edit these titles by hand for one-time or two-time contributors; but you've done a lot of good PRs and it looks like you'll probably do more :green_heart: so I thought I'd ask.

(Note to self: add GH Action to check PR title for conventional commit)

**Comment by Rukario** [comment]
> (Though I did notice that in the label pills, to my eyes, it could be a bit more centered, vertically.

I thought GitHub have gotten the center sorted out that I can replicate - I was on Safari reading this & looking at how GitHub does to their label pills, having gone back to Firefox to open devtools but seem there's nothing special about it, GitHub's also not centered there either (neither does Chrome I sometime spin up for web design discrepancy tests). I knew HTML elements would look different across browsers and I'd notice uneven elements easily but for this labels I failed to (I wasn't bothered to) notice it's not centered vertically until you mention it. It may be a small distraction, not sure if a fix for it would be worthwhile.

@ckerr

I'll be sure to check this out! This looks like good practice to organize PRs for changelog.

File: web/assets/css/transmission-app.scss
Comment (Rukario) [reply web/assets/css/transmission-app.scss:1]: It's a great discovery because you aren't hearing of it ;)

**Comment by GaryElshaw** [comment]
Padding?

**Comment by Rukario** [comment]
I have low confidence to include brute force CSS like that, you'll never know what it might look in other browsers!

**Comment by GaryElshaw** [comment]
> I have low confidence to include brute force CSS like that, you'll never know what it might look in other browsers!

Worth a shot, try it and see. Other options?

**Comment by ckerr** [comment]
> > I have low confidence to include brute force CSS like that, you'll never know what it might look in other browsers!
> 
> Worth a shot, try it and see. Other options?

I don't know if I agree?

Getting CSS just right on all the various browsers out there is tricky. "Worth a shot, try it and see" is just seven words but the amount of work it implies for testing on [platform x browser x form-factor] is pretty complex

**Comment by Rukario** [comment]
> > I have low confidence to include brute force CSS like that, you'll never know what it might look in other browsers!
> 
> Worth a shot, try it and see. Other options?

I have already suggested from my other comment that the GitHub's pill shaped labels looked fine under Safari, but just so I can save your time, I did tried what you have suggested and this is what Safari shows, padded below with just 1 pixel.

IMG_0626

It might look good for lowercase but I'm sure you know lowercase should normally be sag in a div, even GitHub is doing it.

**Pull request merged**